libkbfs/mdserver_remote.go

```diff
@@ -788,12 +795,15 @@ func (md *MDServerRemote) getFoldersForRekey(ctx context.Context,
 
 // Shutdown implements the MDServer interface for MDServerRemote.
 func (md *MDServerRemote) Shutdown() {
+	md.connMu.Lock()
+	defer md.connMu.Unlock()
+
 	// close the connection
 	md.conn.Shutdown()
 	// cancel pending observers
 	md.cancelObservers()
 	// cancel the ping ticker
-	md.resetPingTicker(0)
+	md.pinger.resetTicker(0)
```

Comment:
Now that we've abstracted out the ticker, maybe instead of special-casing a 0-valued `resetTicker`, we can have an `md.pinger.cancelTicker()`. IMO it makes more sense anyway for `pinger.resetTicker(0)` to still ping once, but then not repeat.